### PR TITLE
 .coafile: Add ShellCheckBear

### LINKED ISSUE #2320: Add ShellCheckBear to .coafile
https://github.com/coala/coala/issues/4952

### PR DESCRIPTION
**Pull request by sangamcse**
This add a `bash` section and `bash`
as default `shell` in `.coafile`.

Closes #2320

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `corobo mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by sangamcse** [comment]
@refeed, Thank you for your review. As you can see the commit body's comment, I have mentioned that the file inside .ci was modified as per `shfmt` and not using `shellcheck`. So I have splitted these commit.
Also, check [coala/coala#5243 commits](https://github.com/coala/coala/pull/5243/commits)

File: .ci/package.json.jinja2 [hunk 1]
```diff
@@ -3,4 +3,3 @@
   "version": "{{version}}",
   "dependencies": {{dependencies | jsonify(indent=4, sort_keys=True)}}
 }
-
```
[the remark this answers is not in the document]
Comment (sangamcse) [reply .ci/package.json.jinja2]: Yes. Should I make it newcomer issue or keep it here? I found this while I was modifying shell scripts in `.ci`.

[the remark this answers is not in the document]
Comment (refeed) [reply .ci/package.json.jinja2]: yup, I think a newcomer issue would be fine for this

[the remark this answers is not in the document]
Comment (sangamcse) [reply .ci/package.json.jinja2]: Done https://github.com/coala/coala-bears/issues/2388

**Comment by sangamcse** [comment]
@refeed, good catch 😉. I just removed that line before your comment.

File: .ci/deps.r.sh [hunk 2]
```diff
@@ -2,7 +2,7 @@ set -e
 set -x
 
 # R commands
-echo '.libPaths( c( "'"$R_LIB_USER"'", .libPaths()) )' >> .Rprofile
-echo 'options(repos=structure(c(CRAN="http://cran.rstudio.com")))' >> .Rprofile
+echo '.libPaths( c( "'"$R_LIB_USER"'", .libPaths()) )' >>.Rprofile
```
Comment (jayvdb) [inline .ci/deps.r.sh]: hmm. I find this style very jarring.
can it be disabled?
@yukiisbored your opinion on this rule?

File: .ci/check_maintainership.sh [hunk 3]
```diff
@@ -8,6 +8,6 @@ source ../rultor_secrets.sh
```
[the remark this answers is not in the document]
Comment (jayvdb) [reply .ci/check_maintainership.sh]: That is fixed; rebase and this file disappears

File: .ci/deps.r.sh (hunk 2 above)
Comment (sangamcse) [reply .ci/deps.r.sh]: I don't have much knowledge about these `R` stuff.

File: .ci/deps.pip.sh [hunk 4]
```diff
@@ -5,13 +5,13 @@ TERM=dumb
 
 # Choose the python versions to install deps for
 case $CIRCLE_NODE_INDEX in
- 0) dep_versions=( "3.4.3" "3.5.1" ) ;;
```
[the remark this answers is not in the document]
Comment (sangamcse) [reply .ci/deps.pip.sh]: Should I keep this file or remove all changes?

File: .ci/deps.r.sh (hunk 2 above)
Comment (jayvdb) [reply .ci/deps.r.sh]: Can you find a rule to ignore these changes?

Comment (sangamcse) [reply .ci/deps.r.sh]: Understood.
Actually, this was not done by `shellcheck` but by the `shfmt`. 
> can it be disabled?

I can revert this changes. 😃

File: .ci/deps.apt.sh [hunk 5]
```diff
@@ -19,7 +19,7 @@ case $CIRCLE_BUILD_IMAGE in
     # Use xenial, needed to replace outdated julia provided by Circle CI
     ADD_APT_UBUNTU_RELEASE=xenial
     # Work around lack of systemd on trusty, which xenial's lxc-common expects
-    echo '#!/bin/sh' | sudo tee /usr/bin/systemd-detect-virt > /dev/null
+    echo '#!/bin/sh' | sudo tee /usr/bin/systemd-detect-virt >/dev/null
```
Comment (jayvdb) [inline .ci/deps.apt.sh]: cant we ignore the rule that causes this?

Comment (sangamcse) [reply .ci/deps.apt.sh]: Yes.

**Comment by jayvdb** [comment]
ack https://github.com/coala/coala-bears/pull/2385/commits/7634540263106f9aeb892cfb9eacfddeed54be5d https://github.com/coala/coala-bears/pull/2385/commits/b650a4b4cc13e145fba28585213725438c52c0dc

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**